kong/router/expressions.lua

```diff
@@ -19,8 +24,34 @@ local PROTOCOLS_OVERRIDE = {
 }
 
 
-local function get_exp_and_priority(route)
+-- net.port => net.dst.port
+local function verify_expression(route)
   local exp = route.expression
+
+  if not exp then
+    return nil
+  end
+
+  if not exp:find("net.port", 1, true) then
+    return exp
+  end
+
+  -- there is "net.port" in expression
+
+  local new_exp = re_gsub(exp, NET_PORT_REG, NET_PORT_REPLACE)
```

Comment:
```suggestion
  local new_exp = re_gsub(exp, NET_PORT_REG, NET_PORT_REPLACE, "joi")
```

Reply:
According to ngx_lua's doc, we should not use 'o' option in `ngx.re.gsub`. (https://github.com/openresty/lua-nginx-module#lua_regex_cache_max_entries).

And I think that it will not be a hot path.

Reply:
@chronolaw I think we should, the doc says:

> Do not activate the o option for regular expressions (and/or replace string arguments for [ngx.re.sub](https://github.com/openresty/lua-nginx-module#ngxresub) and [ngx.re.gsub](https://github.com/openresty/lua-nginx-module#ngxregsub)) **that are generated on the fly**...

`NET_PORT_REG` is clearly not generated on the fly, so `o` would be appropriate here.

Reply:
Also, `jo` would be fine, we want to be case-sensitive (no `i`).

Reply:
done, adding "jo"